### PR TITLE
Add monitoring and sylog optionally

### PR DESCRIPTION
**Pull request by hannesm**
Previously, there was the `future` branch in this repository which added the monitoring elements.

An update of the source code included: update the main branch, rebase the future branch on the new main branch, force-push the future branch.

This also meant that some revs aren't available anymore (due to the force-push), and a rather intricate change.

This PR modifies the config.ml to have a configure-time `--monitoring` flag, which, if provided, adds logs-syslog and mirage-monitoring dependency and initialization code. The CI has been modified to run once without --monitoring and once with --monitoring.

WDYT @reynir @dinosaure? If this is fine, I'm keen to adapt other unikernels in the same way.

### CODE REVIEW

**Comment by reynir** [comment]
I think it is very nice. I found that building after `mirage configure --enable-monitoring` on a non-solo5 target lead to a linking error. After a lot of experiments I found a way to make `mirage configure` print an error if `--enable-monitoring` is passed on a non-solo5 target.

I tried and failed to devise a way to hide `--enable-monitoring` on non-solo5 targets.

**Comment by hannesm** [comment]
Thanks @reynir. I wonder whether the mirage-monitoring should remove its mirage-solo5 dependency (and push the `memory_metrics` into `mirage-solo5` directly (the mirage-xen could as well have that)!?

This way, the memory profiling would work on all platforms at least...

**Comment by hannesm** [comment]
But actually, the underlying issue is: Solo5_os.Memory is very mirage-solo5 specific (there's no general interface for the OS module). In addition, the `memory_metrics` has a `~quick:bool` argument (i.e. it is configurable), and it needs to be registered with (`Metrics_lwt.periodically`).

My proposal:
- in mirage-solo5 we add memory_metrics and register it automatically (using `quick_stat`) to the Metrics_lwt.periodically
- in mirage-xen the same
- for the need of a `stat` (i.e. no `quick_stat`), this can always be added manually.

About `Metrics_lwt.periodically`: this is a (mutable) list that can only ever grow. Unless something calls `init_periodic`, there's no work done. I.e. a "normal" unikernel using mirage-solo5/mirage-xen will not be forced to do any work with the above.

So, TL;DR: WDYT about https://github.com/mirage/mirage-solo5/pull/94 https://github.com/mirage/mirage-xen/pull/51? https://git.robur.io/robur/mirage-monitoring/pulls/2 would then remove the mirage-solo5 dependency from mirage-monitoring.

**Comment by reynir** [comment]
I think that would be excellent. Thanks! And if for some reason someone  doesn't want memory reporting they can disable it using `Metrics.disable_tag "memory"`.

**Comment by hannesm** [comment]
@reynir I reverted your commits, since mirage-monitoring 0.0.4 is now out that works fine with mirage-xen and mirage-unix :)

**Comment by reynir** [comment]
Great! I think we should merge these PRs then :)

(I got an email that you pushed 0 commits)

**Comment by hannesm** [comment]
@reynir I'm doing this as we speak. I actually force-pushed -2 commits ;)
